src/snapred/backend/data/LocalDataService.py

```diff
@@ -961,6 +964,24 @@ def writeWorkspace(self, path: Path, filename: Path, workspaceName: WorkspaceNam
         saveAlgo.setProperty("Filename", str(path / filename))
         saveAlgo.execute()
 
+    def writeRaggedWorkspace(self, path: Path, filename: Path, workspaceName: WorkspaceName):
+        """
+        Write a ragged workspace to disk in a .tar format.
+        """
+        saveAlgo = AlgorithmManager.create(WrapLeftovers.__name__)
+        saveAlgo.setProperty("InputWorkspace", workspaceName)
+        saveAlgo.setProperty("Filename", str(path / filename))
+        saveAlgo.execute()
+
+    def readRaggedWorkspace(self, path: Path, filename: Path, workspaceName: WorkspaceName):
+        """
+        Read a ragged workspace from disk in a .tar format.
+        """
+        loadAlgo = AlgorithmManager.create(ReheatLeftovers.__name__)
+        loadAlgo.setProperty("Filename", str(path / filename))
+        loadAlgo.setProperty("OutputWorkspace", workspaceName)
+        loadAlgo.execute()
```

Comment:
this needs a test

Reply:
[?](https://github.com/neutrons/SNAPRed/pull/281/files#diff-d9ed9d591fbbe1d951fb39decc2f8519c788c21334cff233020b7d143989b03bR1606-R1607)

Reply:
Sorry, CodeCov was marking it as untested